src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -240,6 +244,16 @@ class PrisonerIepLevelReviewService(
     )
   }
 
+  private suspend fun sendEventAndAudit(iepDetail: IepDetail) {
+    snsService.sendIepReviewEvent(iepDetail.id!!, iepDetail.iepTime)
```

Comment:
I do wonder how a record could **_not_** have an ID just after the repository has saved it?

I'm wary of these `!!` because they could raise a `NullPointerException`, maybe the compiler was complaining about `id` being a `String?` instead of a `String`?

I suppose this is more of a very very exceptional case (e.g. shouldn't really happen) - however if it is kinda possible for the ID to be `null` after the repository `save()` call and assuming an exception wasn't raised by `save()` itself, it may be worth checking the `null` case explicitly and log a nice message and/or raise a custom exception...

...but I think in this case the result of `save()` should "always" have an `id` right?

Reply:
Good point! Have wrapped that in `let`